app/controllers/projects_controller.rb

```diff
@@ -206,6 +206,8 @@ def projectcommit
   #todo - /tree/master/file_name and generalize for /tree/branch/file_name
 
   def masterbranch
+    authorize! :update_image, @project
```

Comment:
I don't see a reason for this being here... This prevents everyone from entering into the image page(I actually tried it locally and it really blocked other users).. I'd just prevent them from seeing the update button.
So, remove this line.